gcp-pubsub/src/main/java/io/micronaut/gcp/pubsub/bind/PubSubBodyBinder.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.gcp.pubsub.bind;
+
+import com.google.pubsub.v1.PubsubMessage;
+import io.micronaut.core.convert.ArgumentConversionContext;
+import io.micronaut.core.type.Argument;
+import io.micronaut.gcp.pubsub.exception.PubSubListenerException;
+import io.micronaut.gcp.pubsub.serdes.PubSubMessageSerDes;
+import io.micronaut.gcp.pubsub.serdes.PubSubMessageSerDesRegistry;
+import io.micronaut.messaging.annotation.Body;
+
+import javax.inject.Singleton;
+import java.util.Optional;
+
+/**
+ * Binds arguments annotated with {@link Body} and uses the appropriate
+ * {@link io.micronaut.gcp.pubsub.serdes.PubSubMessageSerDes} to deserialize the contents of the PubSubMessage data.
+ *
+ * @author Vinicius Carvalho
+ * @since 2.0.0
+ */
+@Singleton
+public class PubSubBodyBinder implements PubSubAnnotatedArgumentBinder<Body> {
```

Comment:
The body annotation has support for defining a subsection of the body to bind to. I don't know if that makes sense to support in this context.

Reply:
That could be an improvement, can I get a raincheck on this one :)